[Sitemaps] Google Sitemap PageMap extensions, implements #388

sebastian-nagel opened the pull request:
Implementation of Google's PageMap extension, see #388.

rzo1 commented on src/main/java/crawlercommons/sitemaps/sax/extension/PageMapsHandler.java:
```diff
@@ -0,0 +1,114 @@
+ */
+public class PageMapsHandler extends ExtensionHandler {
+
+    PageMap currPageMap;
```
Is there a specific reasons for that fields being package-private?

sebastian-nagel replied:
No. Only my laziness. Will make it class-private.

rzo1 commented on src/main/java/crawlercommons/sitemaps/sax/extension/PageMapsHandler.java:
```diff
@@ -0,0 +1,114 @@
+        currAttrName = null;
+        currDataObj = null;
+        currAttrName = null;
+        currAttrVal = new StringBuilder();
```
Don't know if we are in a critical code path here, but if we don't want to re-create the object, a `setLength(0)` would be sufficient.

sebastian-nagel replied:
Ok.

rzo1 commented on src/main/java/crawlercommons/sitemaps/sax/extension/PageMapsHandler.java:
```diff
@@ -0,0 +1,114 @@
+    }
+
+    private void resetCurrent() {
+        currAttrName = null;
```
This is a duplicate. `currAttrValFromAttr`is missing here.

sebastian-nagel replied:
Good catch! `currAttrValFromAttr` is set to null a few lines below.

sebastian-nagel commented:
Thanks, @rzo1! I've addressed all your comments and also ensured that no NPEs happen with not properly nested PageMap data, eg., DataObject without surrounding PageMap.

rzo1 commented on src/main/java/crawlercommons/sitemaps/extension/PageMapDataObject.java:
```diff
@@ -0,0 +1,85 @@
+
+    @Override
+    public Map<String, String[]> asMap() {
+        // TODO Auto-generated method stub
```
Is this something we should fix?

sebastian-nagel replied:
Yes. I've moved parts of the asMap() method of the PageMap class to here.

rzo1 commented on src/main/java/crawlercommons/sitemaps/sax/extension/PageMapsHandler.java:
```diff
@@ -0,0 +1,119 @@
+    private PageMap currPageMap;
+    private PageMapDataObject currDataObj;
+    private String currAttrName;
+    private StringBuilder currAttrVal = new StringBuilder();;
```
Double ";"

sebastian-nagel replied:
Done.

rzo1 commented on src/test/java/crawlercommons/sitemaps/extension/PageMapsTest.java:
```diff
@@ -0,0 +1,56 @@
+package crawlercommons.sitemaps.extension;
```
missing license header

sebastian-nagel replied:
Added.

sebastian-nagel commented on src/test/resources/sitemaps/extension/pagemaps-sitemap.xml:
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Don't know, the other sample files in this directory also do not have license headers.

rzo1 commented:
lgtm!

sebastian-nagel commented:
Squashed commits and rebased to master to include the fix for #441 affecting the automatic builds.

jnioche commented:
thanks @sebastian-nagel and @rzo1
